static/js/src/ua-product-selector.js

```diff
@@ -0,0 +1,165 @@
+import { StateManager } from "./utils/state.js";
+
+function productSelection() {
+  const form = document.querySelector(".js-shop-form");
+  const steps = ["type", "quantity", "version", "support", "add"];
+  const stepClassPrefix = ".js-shop-step--";
+  let productState = new StateManager(steps, render);
+
+  init();
+  render();
+
+  function init() {
+    const productInputs = form.querySelectorAll(".js-product-input");
+    const publicCloudInputs = form.querySelectorAll(".js-public-cloud-input");
+    const versionTabs = form.querySelectorAll(
+      ".js-shop-step--version .p-tabs__link"
+    );
+
+    productInputs.forEach((input) => {
+      input.addEventListener("input", (e) => {
+        handleStepSpecificAction(e.target);
+      });
+    });
+
+    publicCloudInputs.forEach((input) => {
+      input.addEventListener("input", () => {
+        disableSteps(["quantity", "version", "support", "add"]);
+      });
+    });
+
+    versionTabs.forEach((tab) => {
+      tab.addEventListener("click", (e) => {
+        e.preventDefault();
+        productState.set("version", [tab.getAttribute("href")]);
+      });
+    });
+  }
+
+  function disableSteps(steps) {
+    steps.forEach((step) => {
+      const wrapper = form.querySelector(`${stepClassPrefix}${step}`);
+
+      wrapper.classList.add("u-disable");
+    });
+  }
+
+  function enableSteps(steps) {
+    steps.forEach((step) => {
+      const wrapper = form.querySelector(`${stepClassPrefix}${step}`);
+
+      wrapper.classList.remove("u-disable");
+    });
+  }
+
+  function handleStepSpecificAction(inputElement) {
+    const step = inputElement.name;
+    const validQuantity = step === "quantity" && inputElement.value > 0;
```

Comment:
You dont need the `step === "quantity"` check here as your filtering it in the switch case below